athena-jdbc/src/main/java/com/amazonaws/athena/connectors/jdbc/manager/JDBCUtil.java

```diff
@@ -128,4 +131,20 @@ public static Map<String, JdbcRecordHandler> createJdbcRecordHandlerMap(final Ma
 
         return recordHandlerMap.build();
     }
+
+    public static String checkEnvironment(String databaseEngine, String url)
+    {
+        if ("synapse".equals(databaseEngine)) {
+            // checking whether it's Azure serverless environment or not based on host name
+            Matcher m = Pattern.compile("([a-zA-Z]+)://([^;]+);(.*)").matcher(url);
```

Comment:
It won't be compiled every time as we already made a condition to do that in case of only Synapse.
But as a standard I've made the necessary changes.